File: Hackerrank/30-Days/day-25/solution.js
function processData(input) {
  //Enter your code here
  var data = input.split("\n");
  //printPrime(data[1]);
  for (var i=1;i<data.length;i++){
    //console.log(data[i]);
    printPrime(data[i]);
  }
} 

function printPrime(input) {
  if (input == 1){
      console.log("Not prime");
      return;
  }
  
  var limit = Math.ceil(Math.sqrt(input));
  //console.log('limit:'+limit);
  for(var i=2;i <= limit; i++){
      if (input != i && input % i == 0){
          console.log("Not prime");
          return;
      }
  }
  console.log("Prime");
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
  _input += input;
});

process.stdin.on("end", function () {
 processData(_input);
});

File: Hackerrank/30-Days/day-25/solution.java
import java.io.*;
import java.util.*;
import java.lang.Math; 

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        Scanner scan = new Scanner(System.in); 
        int size = Integer.parseInt(scan.nextLine());
        for(int i=0;i<size;i++){
            int data = Integer.parseInt(scan.nextLine());
            printPrime(data);
        }
        scan.close();
    }
    
    public static void printPrime(int data){
        //System.out.println(data);
        if (data == 1){
            System.out.println("Not prime");
            return;
        }
        
        int limit = (int) Math.ceil(Math.sqrt(data));
        //System.out.println(limit);
        for (int i=2;i<=limit;i++){
            if (data != i && data % i == 0){
                System.out.println("Not prime");
                return;
            }
        }
        
        System.out.println("Prime");
    }
}